tools/c7n_gcp/c7n_gcp/actions/notify.py

```diff
@@ -0,0 +1,153 @@
+            'type': {'enum': ['notify']},
+            'to': {'type': 'array', 'items': {'type': 'string'}},
+            'owner_absent_contact': {'type': 'array', 'items': {'type': 'string'}},
+            'to_from': ValuesFrom.schema,
+            'cc': {'type': 'array', 'items': {'type': 'string'}},
+            'cc_from': ValuesFrom.schema,
+            'cc_manager': {'type': 'boolean'},
+            'from': {'type': 'string'},
+            'subject': {'type': 'string'},
+            'template': {'type': 'string'},
+            'transport': {
+                'oneOf': [
+                    {'type': 'object',
+                     'required': ['type', 'topic'],
+                     'properties': {
+                         'topic': {'type': 'string'},
+                         'type': {'enum': ['pubsub']},
+                     }}],
+            },
+        }
+    }
+
+    @staticmethod
+    def register_notify_action(registry, _):
+        for resource in registry.keys():
+            klass = registry.get(resource)
+            klass.action_registry.register('notify', Notify)
+
+    def process(self, resources, event=None):
+        self.session = utils.local_session(self.manager.session_factory)
+        project = self.session.get_default_project()
+        message = {
+            'event': event,
+            'account_id': project,
+            'account': project,
+            'region': 'all',
+            'policy': self.manager.data
+        }
+
+        message['action'] = self.expand_variables(message)
+
+        for batch in utils.chunks(resources, self.batch_size):
+            message['resources'] = batch
+            receipt = self.send_data_message(message)
+            self.log.info("sent message:%s policy:%s template:%s count:%s" % (
+                receipt, self.manager.data['name'],
+                self.data.get('template', 'default'), len(batch)))
+
+    def send_data_message(self, message):
+        return self.publish_message(message)
+
+    # Methods to handle GCP Pub Sub topic publishing
+    def publish_message(self, message):
+        """Publish message to a GCP pub/sub topic
+         """
+        topic = self.ensure_topic()
+
+        client = self.session.client('pubsub', 'v1', 'projects.topics')
+
+        try:
+            return client.execute_command('publish', {
+                'topic': topic,
+                'body': {
+                    'messages': {
+                        'data': self.pack(message)
+                    }
+                }
+            })
+        except HttpError as e:
+            if e.resp.status != 404:
+                raise
+
+    def get_topic_param(self, topic=None, project=None):
+        """Returns Rest API URI formatted with topic and project in it
+        """
+        return 'projects/{}/topics/{}'.format(
+            project or self.session.get_default_project(),
+            topic or self.data['transport']['topic'])
+
+    def ensure_topic(self):
```

Comment:
this seems to be copy and paste from c7n_gcp/mu.py i'd suggest dropping it there's no point in auto creating the topic, the user has to setup something on the other side to have delivery.

Reply:
I was following what I thought was a pattern in mu to make sure the topic was always there, assuming the user didn't set it up. was waiting for your feedback on this as to whether to actually keep it or not. Next commit will have this removed.

Reply:
mu is hooking up things end to end for event delivery to the policy function, notify cant and shouldn't be the, since it doesn't control the consumer. the user has to define this ahead of time.

Reply:
Understood bro.